near-secret-agent/agent.py

```diff
@@ -0,0 +1,120 @@
+"""
+MODEL = "qwen2p5-72b-instruct"
+
+class Agent:
+    def __init__(self, env):
+        self.env = env
+        self.revealed = False
+        self.x_consumer_key = self.env.env_vars.get("X_CONSUMER_KEY", None)
+        if not self.x_consumer_key:
+            raise ValueError("X_CONSUMER_KEY is required")
+        self.x_consumer_secret = self.env.env_vars.get("X_CONSUMER_SECRET", None)
+        if not self.x_consumer_secret:
+            raise ValueError("X_CONSUMER_SECRET is required")
+        self.x_access_token = self.env.env_vars.get("X_ACCESS_TOKEN", None)
+        if not self.x_access_token:
+            raise ValueError("X_ACCESS_TOKEN is required")
+        self.x_access_token_secret = self.env.env_vars.get("X_ACCESS_TOKEN_SECRET", None)
+        if not self.x_access_token_secret:
+            raise ValueError("X_ACCESS_TOKEN_SECRET is required")
+        self.x_client = tweepy.Client(
+            consumer_key=self.x_consumer_key,
+            consumer_secret=self.x_consumer_secret,
+            access_token=self.x_access_token,
+            access_token_secret=self.x_access_token_secret
+        )
+
+    def reveal_secrets(self):
+        """Reveal the secrets of NEARvana"""
+        # self.env.env_vars.get("TODAYS_SECRET", "NEAR is the blockchain for AI")
+        self.revealed = True
+
+    def validate_hub_user(self, auth):
+        """Only accept Twitter messages if they come from the hub user"""
+        parsed = json.loads(auth)
+        if parsed["account_id"] != "flatirons.near": # todo replace with hub user
+            raise ValueError("Unauthorized")
+
+    def tweet_reply(self, event, reply):
+        """Reply to a tweet"""
+        tweet_id = event["tweet_id"]
+
+        # Reply to the tweet
+        response = self.x_client.create_tweet(text=reply, in_reply_to_tweet_id=tweet_id)
+
+        print("Reply sent successfully:", response)
+
+
+    def run(self):
+        env = self.env
+        self.validate_hub_user(env.client._auth)
+        last_message = env.list_messages()[-1]
+
+        print(last_message)
+        if last_message is None:
+            print("No message found")
+            return
+
+        if not last_message["content"]:
+            print("Message content was empty")
+            return
+
+        event = json.loads(last_message["content"])
+        tweet_id = event["tweet_id"]
+        tweet_key = f"tweet-status-{tweet_id}"
+
+        existing = env.get_agent_data_by_key(tweet_key)
+        if existing:
+            if existing["value"] == "complete" or existing["value"] == "error":
+                print(f"Tweet {tweet_id} already processed")
+                return
+            else:
+                # todo resume processing. Try once then mark as error
+                pass
+
+        env.save_agent_data(tweet_key, "processing")
+
+        tool_registry = env.get_tool_registry(True)
+        tool_registry.register_tool(self.reveal_secrets)
```

Comment:
How this tool suppose to work? Could you add more details how to manage secrets if this code is ready to be merged.